extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -39,6 +39,8 @@ export class AmpExperiment extends AMP.BaseElement {
     if (!this.isExperimentOn_) {
       dev().warn(EXPERIMENT, `Experiment ${EXPERIMENT} disabled`);
       toggle(this.element, false);
+      getService(this.win, 'variant',
+          () => Promise.reject(EXPERIMENT + ' disabled'));
```

Comment:
Good point. `Promise.all` fails fast. The only reason I used `reject` is to give a better error message. I can change it to `resolve`.